### PR TITLE
Responsive Mobile Design for Detail Tabs #4a9cd1

### PR DESCRIPTION
**Pull request by mahamzayed**
# Description

The purpose of this PR is to add mobile responsive functionality for the tabs section of the detail pages according to the [designs](https://app.abstract.com/projects/e709ca90-5ba9-11e9-9eaa-59df656d954a/branches/600d2375-3dd1-4940-97a6-bbec7a21f97e/commits/2588b1accae888999497eac9165f0f8346c255f1/files/a878d736-29fb-4fd2-b67e-dbb681e9f1d9/layers/E3503EFE-0CF8-4A2E-B251-EE6DB96C2756?collectionId=4057666d-814a-4cec-a2e9-58c3b5b8e1ec&mode=build&selected=1545286161-F88DBCD5-3AD4-48BE-AEB1-69AF51376791).

Please note the following:
- In the design, the detail tabs are faded towards the end to suggest that you need to keep scrolling more to see them on mobile. This has been changed to a different pattern, in which the tabs are partially revealed at the end to indicate that you should keep scrolling on mobile. You can test this by adding more tabs to the list in Vue dev tools to see if that pattern is being applied.
- The size of the banner image on detail pages for sizes 320px and lower has been reduced to 288px to match the designs.
- Descriptions should be truncated to 270 words on mobile and titles should be truncated to 90 words as per design request.
- Any viewport that is 700px and up will have a tablet-like viewport, while anything lower than 700px will lean towards the mobile viewport.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

View all detail pages in the mobile viewport (767px and lower). Pages should match the mobile designs.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by mahamzayed** [comment]
@cameronbaney Let me know if you run into any design inconsistencies. I tried to address as many edge cases as I could across all detail pages and inspect them across different viewports other than mobile.

File: components/DetailTabs/DetailTabs.vue [hunk 1]
```diff
@@ -3,18 +3,18 @@
...
+                <a
+                  class="details-tabs__container--button"
+                  :class="{ active: tab.type === activeTab }"
+                  href="#"
+                  @click.prevent="$emit('set-active-tab', tab.type)"
+                >
+                  {{ tab.label }}
+                </a>
+              </li>
+            </ul>
```
Comment (matthewtgilbride) [inline components/DetailTabs/DetailTabs.vue]: this looks like it's just an accidental extra tab

File: components/DetailTabs/DetailTabs.vue [hunk 2]
```diff
@@ -113,10 +113,34 @@ export default {
...
+      }
+    }
+  }
+
+  ::v-deep .el-row--flex {
+    overflow-y: scroll;
+  }
+}
+
+@media screen and (max-width: 1023px) {
```
Comment (matthewtgilbride) [inline components/DetailTabs/DetailTabs.vue]: lol, assuming this is similar to what we were chatting about yesterday - you want to target everything _up to_ but not including 1024 (Ipad Pro or original Ipad landscape), correct?

Just making sure its not a typo.

File: components/DetailsHeader/DetailsHeader.vue [hunk 3]
```diff
@@ -51,21 +57,49 @@ export default {
...
     /**
-     * Formats title length
+     * Formats breadcrumb length
+     * @param {String} breadcrumb
+     */
+    formatBreadcrumb: function(breadcrumb) {
+      return breadcrumb.length > 32
+        ? breadcrumb.substring(0, 32) + '...'
+        : breadcrumb
+    },
```
Comment (matthewtgilbride) [inline components/DetailsHeader/DetailsHeader.vue]: I think this is unused now that we have a breadcrumb component and can be removed.

Comment (mahamzayed) [reply components/DetailsHeader/DetailsHeader.vue]: @matthewtgilbride We have a breadcrumb component? I wasn't aware lol let me check it out

File: components/DetailTabs/DetailTabs.vue (hunk 2 above)
Comment (mahamzayed) [reply components/DetailTabs/DetailTabs.vue]: Yessir! The goal is to have that gray line underneath the tabs section always extend to the end of the container starting from mobile all the way up to and including Ipad viewports. If 1024 includes Ipad Pro/Landscape, I can change to reflect that - just one px difference :)

**Comment by mahamzayed** [comment]
@cameronbaney Thanks Cameron! I'll see if I can find an alternative solution instead of using two headings and two descriptions.

**Pull request merged**